servers/audio_server.cpp

```diff
@@ -1032,6 +1048,14 @@ void AudioServer::update() {
 	for (Set<CallbackItem>::Element *E = update_callbacks.front(); E; E = E->next()) {
 		E->get().callback(E->get().userdata);
 	}
+
+	if (Engine::get_singleton()->is_editor_hint()) {
```

Comment:
If this change should only work for editor, maybe `#ifdef TOOLS_ENABLED` should also be added, so it's not included in templates builds?

Reply:
The main reason I see to do that is export binary size and potentially (slightly) reducing lock contention, which makes me wonder if I should do it elsewhere too. What do you think?

Reply:
I haven't yet checked the whole PR, but I can already say if you want something to happen only in the editor the current check is better, because it is `false` both on tools builds when running a project (vs. running the editor) and on non-tools builds.
Furthermore, in non-tools builds, given the check is known to fail at compile time, I'm quite certain that compilers can optimize out the body.

Reply:
Well, some places seem to use only `Engine::get_singleton()->is_editor_hint()` while others use both `Engine::get_singleton()->is_editor_hint()` and `TOOLS_ENABLED` checks. Maybe I'm missing something, but if using only `Engine::get_singleton()->is_editor_hint()` is preferred it's good to know :)

Reply:
There's no clear policy, but yes we sometimes put both checks in place to ensure that the editor-specific code (often expensive debug code) is compiled out in the runtime. If @RandomShaper is correct that compiler would do it just fine with `is_editor_hint()` only, then we shouldn't need to do that indeed.

Reply:
If we want to have access to this feature enabled for any other engine mode in project settings, then I guess `TOOLS_ENABLED` isn't needed too.